### DISCUSSION TITLE
Solving multi-objective planning with Optuna, not limited to hyperparametric optimisation?

### DISCUSSION THREAD

**Discussion by Sanatus** [optuna/optuna#5625] [Q&A]
Optuna can set multiple OBJECTIVES and to parameters to achieve multi-objective optimisation, so if I customise more than one objective function, can I solve the multi-objective planning problem, but I don't know how to limit the scope of the constraint function, for examples:

def objective(trial):
    x = trial.suggest_float("x", -5, 5)
    y = trial.suggest_float("y", -5, 5)
    r = trial.suggest_int("r", 1, 6)
    q = trial.suggest_int("q", 1, 4)
    ads = x*r - q
    objective_1 = x**2 + y**2 + (x - 2) * (ads >=-1)
    objective_2 = (x - 2)**2 + (y - 2)**2 + (ads-6) * r
 
    return objective_1, objective_2

Where ads is my constraint function, but I can't seem to put a constraint on it, if I make ads = x*r - q <= 10, it will run, but it doesn't seem to act as a constraint.
Is there a solution?

**Comment by nzw0301** [comment]
I'm wondering if you could use an approach described in this FAQ section https://optuna.readthedocs.io/en/stable/faq.html#how-can-i-optimize-a-model-with-some-constraints?
